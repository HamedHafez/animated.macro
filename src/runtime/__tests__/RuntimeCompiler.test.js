const compile = require("../");
const { Animated } = require("react-native");
const { Node: AnimatedNode } = Animated;

describe("RuntimeCompiler", () => {
  test("cast literal to AnimatedNode", () => {
    expect(compile`1`).toBeInstanceOf(AnimatedNode);
  });

  test("cast numeric placeholder to AnimatedNode", () => {
    expect(compile`${1}`).toBeInstanceOf(AnimatedNode);
  });

  test("pass through Animated.Value placeholder", () => {
    const value = new Animated.Value(10);
    expect(compile`${value}`).toBe(value);
  });

  test("adding two placeholders", () => {
    const value = new Animated.Value(10);
    const compiled = compile`${value} + ${value}`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });

  test("adding two literals", () => {
    const compiled = compile`1 + 2`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });

  test("modulo doesn't cast rhs to Animated", () => {
    const compiled = compile`5 % 3`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();

    // NOTE: Depends on an implementation detail of AnimatedModulo
    expect(compiled._modulus).toBe(3);
  });

  test("inverting a placeholder", () => {
    const value = new Animated.Value(10);
    const compiled = compile`1 / ${value}`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });

  test("multiplying a placeholder and a literal", () => {
    const value = new Animated.Value(10);
    const compiled = compile`${value} * 3`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });

  test("multiplying literal and a placeholder", () => {
    const value = new Animated.Value(10);
    const compiled = compile`3 * ${value}`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });

  test("inverting a parenthesized addition", () => {
    const value = new Animated.Value(0);
    const compiled = compile`1 / (${value} + 1)`;
    expect(compiled).toBeInstanceOf(AnimatedNode);
    expect(compiled).toMatchSnapshot();
  });
});
