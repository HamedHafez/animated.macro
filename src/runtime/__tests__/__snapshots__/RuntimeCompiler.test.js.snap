// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuntimeCompiler adding two literals 1`] = `3`;

exports[`RuntimeCompiler adding two placeholders 1`] = `20`;

exports[`RuntimeCompiler inverting a placeholder 1`] = `0.1`;

exports[`RuntimeCompiler modulo doesn't cast rhs to Animated 1`] = `2`;

exports[`RuntimeCompiler multiplying a placeholder and a literal 1`] = `30`;

exports[`RuntimeCompiler multiplying literal and a placeholder 1`] = `30`;
